### PR TITLE
Fix function arguments types reported by PHPStan

### PR DESCRIPTION
**Pull request by xknown**
This PR attempts to fix the arguments type passed to a WP core and PHP's built-in functions, which would hopefully help reduce the number of PHP fatals for PHP 8.0.

The full list of reported issues can be seen at

https://gist.github.com/xknown/77f8cbe233da75080d1e9258a8c94f95

https://core.trac.wordpress.org/ticket/51423

### CODE REVIEW

File: src/wp-admin/comment.php [hunk 1]
```diff
@@ -246,7 +252,7 @@
 
 		<?php wp_nonce_field( $nonce_action ); ?>
 	<input type="hidden" name="action" value="<?php echo esc_attr( $formaction ); ?>" />
-	<input type="hidden" name="c" value="<?php echo esc_attr( $comment->comment_ID ); ?>" />
+	<input type="hidden" name="c" value="<?php echo intval( $comment->comment_ID ); ?>" />
```
Comment (jrfnl) [inline src/wp-admin/comment.php:255]:
```suggestion
	<input type="hidden" name="c" value="<?php echo (int) $comment->comment_ID; ?>" />
```

File: src/wp-admin/comment.php [hunk 2]
```diff
@@ -279,7 +288,7 @@
 			comment_footer_die( __( 'Sorry, you are not allowed to edit comments on this post.' ) );
 		}
 
-		if ( wp_get_referer() && ! $noredir && false === strpos( wp_get_referer(), 'comment.php' ) ) {
+		if ( wp_get_referer() && ! $noredir && false === strpos( strval( wp_get_referer() ), 'comment.php' ) ) {
```
Comment (jrfnl) [inline src/wp-admin/comment.php:291]:
```suggestion
		if ( wp_get_referer() && ! $noredir && false === strpos( (string) wp_get_referer(), 'comment.php' ) ) {
```

File: src/wp-admin/customize.php [hunk 3]
```diff
@@ -261,7 +261,7 @@
 							$class .= ' active';
 						}
 						?>
-						<button type="button" class="<?php echo esc_attr( $class ); ?>" aria-pressed="<?php echo esc_attr( $active ); ?>" data-device="<?php echo esc_attr( $device ); ?>">
+						<button type="button" class="<?php echo esc_attr( $class ); ?>" aria-pressed="<?php echo esc_attr( $active ? 'true' : 'false' ); ?>" data-device="<?php echo esc_attr( $device ); ?>">
```
Comment (jrfnl) [inline src/wp-admin/customize.php:264]: If the values are hard-coded strings, `'true'`/`'false'`, the call to `esc_attr()` can be removed.

File: src/wp-admin/admin-header.php [hunk 4]
```diff
@@ -170,8 +170,8 @@
 	$admin_body_class .= ' taxonomy-' . $current_screen->taxonomy;
 }
 
-$admin_body_class .= ' branch-' . str_replace( array( '.', ',' ), '-', (float) get_bloginfo( 'version' ) );
-$admin_body_class .= ' version-' . str_replace( '.', '-', preg_replace( '/^([.0-9]+).*/', '$1', get_bloginfo( 'version' ) ) );
+$admin_body_class .= ' branch-' . str_replace( array( '.', ',' ), '-', (string) ( (float) get_bloginfo( 'version' ) ) );
+$admin_body_class .= ' version-' . str_replace( '.', '-', (string) preg_replace( '/^([.0-9]+).*/', '$1', get_bloginfo( 'version' ) ) );
```
Comment (jrfnl) [inline src/wp-admin/admin-header.php:174]: `preg_replace()` will always return a string when given a string, except when the regex errored out and this regex does not look like on which will error.

File: src/wp-admin/edit-form-advanced.php [hunk 5]
```diff
@@ -242,7 +242,7 @@
 
 $form_action  = 'editpost';
 $nonce_action = 'update-post_' . $post_ID;
-$form_extra  .= "<input type='hidden' id='post_ID' name='post_ID' value='" . esc_attr( $post_ID ) . "' />";
+$form_extra  .= "<input type='hidden' id='post_ID' name='post_ID' value='" . intval( $post_ID ) . "' />";
```
Comment (jrfnl) [inline src/wp-admin/edit-form-advanced.php:245]:
```suggestion
$form_extra  .= "<input type='hidden' id='post_ID' name='post_ID' value='" . (int) $post_ID . "' />";
```

File: src/wp-admin/includes/post.php [hunk 6]
```diff
@@ -1561,7 +1573,7 @@ function _wp_post_thumbnail_html( $thumbnail_id = null, $post = null ) {
 		}
 	}
 
-	$content .= '<input type="hidden" id="_thumbnail_id" name="_thumbnail_id" value="' . esc_attr( $thumbnail_id ? $thumbnail_id : '-1' ) . '" />';
+	$content .= '<input type="hidden" id="_thumbnail_id" name="_thumbnail_id" value="' . intval( $thumbnail_id ? $thumbnail_id : '-1' ) . '" />';
```
Comment (jrfnl) [inline src/wp-admin/includes/post.php:1576]:
```suggestion
	$content .= '<input type="hidden" id="_thumbnail_id" name="_thumbnail_id" value="' . ( $thumbnail_id ? (int) $thumbnail_id : '-1' ) . '" />';
```

**Comment by jrfnl** [comment]
Looked through a few of the changes (by far not all) and would like to observe the following: a lot of these changes bypass underlying errors by doing type casting, instead of handling the errors properly. While this will work, I'm not sure this is the always the right choice.

**Comment by xknown** [comment]
> Looked through a few of the changes (by far not all) and would like to observe the following: a lot of these changes bypass underlying errors by doing type casting, instead of handling the errors properly. While this will work, I'm not sure this is always the right choice.

Thanks for the feedback, and sorry for the late response. I have been busy with some work-related things, and some afk time. I agree that most of the changes are only adding explicit casts, which is probably fine in most cases.

I've been manually reviewing these changes, and while I add some proper error checking in some places, some commits only add explicit casts mostly because the surrounding code don't really handle very well error conditions (one example that comes to my mind is errors happening when running `wp_install()`). If you point me out a few cases where do you think we can do something different, then please let me know.

**Comment by xknown** [comment]
Closing in favor of individual PRs. See https://core.trac.wordpress.org/ticket/51423 for more details.

**Pull request closed without merging**